Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+                    {
+                        string[] userCredentials = reader.ReadLine().Split(';');
+                        if (CheckIfUserExists(userCredentials[0]))
+                        {
+                            throw new DuplicateUserCredentialsException(userCredentials[0]);
+                        }
+                        else
+                        {
+                            AddUserToArrayAtIndex(userCredentials, 0);
+                        }
+                    }
+                }
+            }
+            catch (FileNotFoundException fileEx)
+            {
+            }
+            catch (DuplicateUserCredentialsException usersEx)
+            {
+                throw;
+            }
+            catch (Exception defaultEx)
+            { 
+                throw;
+            }
+
+        }
+
+        // Function to write user info from userArray to users.txt file
+        static void WriteUsersToFile()
+        {
+            try
+            {
+                using (StreamWriter writer = new StreamWriter(pathToFile,false))
+                {
+                    for (int i = 0; i < userArray.Length; i++)
+                    {
+                        writer.WriteLine($"{userArray[i][0]};{userArray[i][1]}");
+                    }
+                }
+            }
+            catch (Exception defaultEx)
+            {
+                throw;
+            }
+        }
+
+        //Funtion that gets which option was selected and calls other needed functions via switch statement.
+        static void OptionSelection(string option) {
+            Console.Clear();
+            switch (option)
+            {
+                case "1":
+                    LogIn();
+                    break;
+                case "2":
+                    PromptAddUserToArray();
+                    break;
+                case "3":
+                    ListUsers();
+                    break;
+                case "4":
+                    PromptAddUserToArray(addToEnd: false, addToStart: true);
+                    break;
+                case "5":
+                    PromptAddUserToArray();
+                    break;
+                case "6":
+                    PromptAddUserToArray(addToEnd: false, addToStart: false);
+                    break;
+                case "7":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: true);
+                    break;
+                case "8":
+                    PromptRemoveUser(removeFromEnd: true, removeFromStart: false);
+                    break;
+                case "9":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: false);
+                    break;
+                case "10":
+                    SortUsers();
+                    break;
+                case "11":
+                    LogOut();
+                    break;
+                case "12":
+                    WriteUsersToFile();
```

Comment:
You call `WriteUsersToFile` every time you add a user to a file and everytime you remove a user from a file so why do you need to call it here?

Reply:
I wanted to have something like autosave so I wouldn't have to add users every time in case I messed up somewhere.